### DISCUSSION TITLE
swupdate on Jetson Nano emmc

### DISCUSSION THREAD

**Discussion by harri-pm** [OE4T/meta-tegra#860] [Q&A] [answered]
Hello,

I post an another question. I thought I got the update working one time with previous instructions https://github.com/OE4T/meta-tegra/discussions/855 and https://github.com/OE4T/meta-tegra/discussions/842. However, I am stuck with an error during update:
```
root@grabber-nano:/part# swupdate -e system,slot_b -H software:1.0 -i swupdate-image-grabber-grabber-nano-20211119093951.swu 
Swupdate v2021.04.0

Licensed under GPLv2. See source distribution for detailed copyright notices.

[INFO ] : SWUPDATE started :  Software Update started !
[INFO ] : SWUPDATE running :  Installation in progress
[ERROR] : SWUPDATE failed [0] ERROR : Error: missing entries for partitions: BCT, RP1, PT, VER, DTB, LNX
[ERROR] : SWUPDATE failed [0] ERROR :        for TNSPEC 3448-401-0002-F.0-1-0-grabber-nano-mmcblk0p1
[ERROR] : SWUPDATE failed [0] ERROR : ERR: bootloader update failed
[ERROR] : SWUPDATE failed [0] ERROR : execute postinstall scripts failed
[ERROR] : SWUPDATE failed [1] Installation failed !
[INFO ] : No SWUPDATE running :  Waiting for requests...
```

My partition layout, image recipe and swupdate recipe are the same or similar as in the previous threads and the machine config file file seems to be the same as in the test distro, at least for the parts that I think are related to the bootloader or partitions. This is first time me working with boot loaders so could you shed some light on this error? Thanks for the answers.

**Comment by madisongh** [comment, marked as answer]
Based on the TNSPEC mentioned in the error message, it looks like your Nano's FAB revision is `401`.  It must be a new revision of the module, and the BUP generator doesn't have an entry for it. Assuming you copied the `jetson-nano-devkit-emmc.conf` file from the layer as the starting point for your custom machine configuration, just add `fab=401` to the `TEGRA_BUPGEN_SPECS` setting in that file.

**Reply by harri-pm** [reply]
Thanks again! That solved it.

It seems to boot to the old partition now, but I'll try what was done here: https://github.com/OE4T/meta-tegra/discussions/842.
